### PR TITLE
Support libprotobuf 4.24.4

### PR DESCRIPTION
**Pull request by dangercrow**
I'd like to use grpcio 1.62.2 which creates a dependency on libprotobuf which is pinned to only 4.25.3

Looking at the upstream pip package, this could be  pinned to >=4.21.3,<5, so support at least some more minor versions

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by dangercrow** [comment]
@conda-forge-admin, please rerender

**Comment by dangercrow** [comment]
@jjerphan thanks for spotting this so quickly! I'm sadly making this PR from a phone, so changes from my end will be quite slow, but any suggestions / edits are welcome

**Comment by dangercrow** [comment]
@conda-forge-admin, please rerender

**Comment by dangercrow** [comment]
I'm unsure if this requires a build number bump due to the rebuild of 4.25 or if it should be left at 0 for 4.24

I assume it would be more preferable to have a unified build 1 than to replace a build 0? I will add that and re-render

**Comment by dangercrow** [comment]
@conda-forge-admin, please rerender

**Comment by dangercrow** [comment]
@jjerphan I believe this is ready for review now!

**Review by jjerphan — APPROVED** [review]
LGTM, but I would wait for at least another maintainer's review.

The C standard library migration can be performed as part of this PR. The guidelines are given in https://github.com/conda-forge/grpc-cpp-feedstock/pull/367#issuecomment-2141736272 .

**Review by h-vetinari — CHANGES_REQUESTED** [review]
Aside from the missing stdlib changes pointed out by the linter, can you please explain why you cannot move on to protobuf 4.25?

File: recipe/conda_build_config.yaml
Comment (dangercrow) [reply recipe/conda_build_config.yaml:6]: At a high level, this is simply in the dependency tree of the protobuf version.

How do you think this should be cited? A chain of dependencies from protobuf to libabseil?

**Comment by dangercrow** [comment]
> Can you please explain why you cannot move on to protobuf 4.25?

I'm using an external proprietary library which cannot move on to protobuf 4.25

Given that the upstream claim https://github.com/grpc/grpc/blob/9741a17371a4785d8cd9a7840f2b2865d7b7f11d/requirements.txt#L4 is `protobuf>=4.21.3,<5.0dev` I would think that any variant in this interval should be non-controversial?

**Comment by dangercrow** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml
Comment (dangercrow) [reply recipe/conda_build_config.yaml]: I'd spotted the original specific version from the branch built artifact -- it doesn't exist in public conda-forge as there is no existing libprotobuf=4.24 artifact

I've added a comment as best I can

**Comment by h-vetinari** [comment]
> Given that the upstream [claim](https://github.com/grpc/grpc/blob/9741a17371a4785d8cd9a7840f2b2865d7b7f11d/requirements.txt#L4) is `protobuf>=4.21.3,<5.0dev` I would think that any variant in this interval should be non-controversial?

Perhaps not controversial, but definitely a burden. We can only reasonably support one protobuf version at a time, because that's the way shared library ABI works. Actually, protobuf makes our lives even harder there, because we have to rebuild per patch version (rather than the feature-level represented by the minor number), otherwise the right libraries/symbols do not get found.

We can make exceptions for libraries in conda-forge that cannot move on yet, but opaque external libraries do not fall under that - it's a slippery slope, where tomorrow someone needs 4.24.3, and then 4.23.5, and so on, and we would need to double/triple/quadruple our CI matrix here. From my POV there needs to be at the very least an in-conda-forge use-case to justify rebuilds for older pins. IMO you should rather spend your effort on making that other library 4.25 compatible.

**Comment by dangercrow** [comment]
> From my POV there needs to be at the very least an in-conda-forge use-case to justify rebuilds for older pins

So, in conclusion, close PR?

**Comment by h-vetinari** [comment]
I don't speak for all of conda-forge, or even all of @conda-forge/grpc-cpp. Perhaps you can convince enough people to outweigh my POV. As an alternative quick fix, you could also use this recipe as the basis for some local builds that you push to a private channel (don't forget to run `conda index` to create the channeldata), where you build grpc 1.62 against libprotobuf 4.24.4.

**Pull request closed without merging**
